### PR TITLE
update aws.go

### PR DESCRIPTION
**Pull request by lovebaby979**
klog

**What this PR does / why we need it**:
Structured Logging migration
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Structured Logging migration
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
Structured Logging migration
```

### CODE REVIEW

File: staging/src/k8s.io/legacy-cloud-providers/aws/aws.go [hunk 1]
```diff
@@ -3468,12 +3467,12 @@ func (c *Cloud) findELBSubnets(internalELB bool) ([]string, error) {
 		// If we have two subnets for the same AZ we arbitrarily choose the one that is first lexicographically.
 		// TODO: Should this be an error.
 		if strings.Compare(*existing.SubnetId, *subnet.SubnetId) > 0 {
-			klog.Warningf("Found multiple subnets in AZ %q; choosing %q between subnets %q and %q", az, *subnet.SubnetId, *existing.SubnetId, *subnet.SubnetId)
+			klog.InfoS("Found multiple subnets in AZ", " AZ ",az, "choosingSubnetId", *subnet.SubnetId, "existingSubnetId", *existing.SubnetId,  "subnetId2", *subnet.SubnetId)
```
Comment (nckturner) [inline staging/src/k8s.io/legacy-cloud-providers/aws/aws.go:3470]: `choosingSubnetId` -> `chosenSubnetId`
`existingSubnetId` -> `firstSubnet`
`subnetId2` -> `secondSubnet`

File: staging/src/k8s.io/legacy-cloud-providers/aws/aws.go [hunk 2]
```diff
@@ -3468,12 +3467,12 @@ func (c *Cloud) findELBSubnets(internalELB bool) ([]string, error) {
...
 			subnetsByAZ[az] = subnet
 			continue
 		}
 
-		klog.Warningf("Found multiple subnets in AZ %q; choosing %q between subnets %q and %q", az, *existing.SubnetId, *existing.SubnetId, *subnet.SubnetId)
+		klog.InfoS("Found multiple subnets in AZ", " AZ ",az, "choosingSubnetId", *existing.SubnetId,"existingSubnetId", *existing.SubnetId,  "subnetId2",*subnet.SubnetId)
```
Comment (nckturner) [inline staging/src/k8s.io/legacy-cloud-providers/aws/aws.go:3475]: `Found multiple subnets in AZ.` -> `Found multiple subnets in AZ. Choosing one.`
`choosingSubnetId` -> `chosenSubnetId`
`existingSubnetId` -> `firstSubnet`
`subnetId2` -> `secondSubnet`

File: staging/src/k8s.io/legacy-cloud-providers/aws/aws.go [hunk 3]
```diff
@@ -3785,7 +3783,7 @@ func (c *Cloud) EnsureLoadBalancer(ctx context.Context, clusterName string, apiS
 	if isLBExternal(annotations) {
 		return nil, cloudprovider.ImplementedElsewhere
 	}
-	klog.V(2).Infof("EnsureLoadBalancer(%v, %v, %v, %v, %v, %v, %v)",
+	klog.V(2).InfoS("EnsureLoadBalancer",
```
Comment (nckturner) [inline staging/src/k8s.io/legacy-cloud-providers/aws/aws.go:3786]: Missing variable names: `clusterName`, `serviceNamespace`, `serviceName`, `region`, `serviceLoadbalancerIp`, `servicePorts`, `serviceAnnotations`.

File: staging/src/k8s.io/legacy-cloud-providers/aws/aws.go [hunk 4]
```diff
@@ -3807,7 +3805,7 @@ func (c *Cloud) EnsureLoadBalancer(ctx context.Context, clusterName string, apiS
 		}
 
 		if port.NodePort == 0 {
-			klog.Errorf("Ignoring port without NodePort defined: %v", port)
+			klog.ErrorS(nil,"Ignoring port without NodePort defined", "nodePort", port)
```
Comment (nckturner) [inline staging/src/k8s.io/legacy-cloud-providers/aws/aws.go:3808]: `nodePort` -> `port`

File: staging/src/k8s.io/legacy-cloud-providers/aws/aws.go [hunk 5]
```diff
@@ -3916,7 +3914,7 @@ func (c *Cloud) EnsureLoadBalancer(ctx context.Context, clusterName string, apiS
 
 		err = c.updateInstanceSecurityGroupsForNLB(loadBalancerName, instances, subnetCidrs, sourceRangeCidrs, v2Mappings)
 		if err != nil {
-			klog.Warningf("Error opening ingress rules for the load balancer to the instances: %q", err)
+			klog.InfoS("Error opening ingress rules for the load balancer to the instances.","error", err)
```
Comment (nckturner) [inline staging/src/k8s.io/legacy-cloud-providers/aws/aws.go:3917]: `klog.ErrorS`?

File: staging/src/k8s.io/legacy-cloud-providers/aws/aws.go [hunk 6]
```diff
@@ -4134,7 +4132,7 @@ func (c *Cloud) EnsureLoadBalancer(ctx context.Context, clusterName string, apiS
 		break
 	}
 	if path, healthCheckNodePort := servicehelpers.GetServiceHealthCheckPathPort(apiService); path != "" {
-		klog.V(4).Infof("service %v (%v) needs health checks on :%d%s)", apiService.Name, loadBalancerName, healthCheckNodePort, path)
+		klog.V(4).InfoS("service needs health checks.", "serviceName", apiService.Name, "loadBalancerName", loadBalancerName, "nodePort", healthCheckNodePort, "path", path)
```
Comment (nckturner) [inline staging/src/k8s.io/legacy-cloud-providers/aws/aws.go:4135]: `nodePort` -> `healthCheckNodePort`

File: staging/src/k8s.io/legacy-cloud-providers/aws/aws.go [hunk 7]
```diff
@@ -4160,17 +4158,17 @@ func (c *Cloud) EnsureLoadBalancer(ctx context.Context, clusterName string, apiS
...
 	if err != nil {
-		klog.Warningf("Error opening ingress rules for the load balancer to the instances: %q", err)
+		klog.InfoS("Error opening ingress rules for the load balancer to the instances.", "error", err)
 		return nil, err
 	}
 
 	err = c.ensureLoadBalancerInstances(aws.StringValue(loadBalancer.LoadBalancerName), loadBalancer.Instances, instances)
 	if err != nil {
-		klog.Warningf("Error registering instances with the load balancer: %q", err)
+		klog.InfoS("Error registering instances with the load balancer.", "error", err)
```
Comment (nckturner) [inline staging/src/k8s.io/legacy-cloud-providers/aws/aws.go:4167]: `klog.ErrorS`?

**Review by nckturner — CHANGES_REQUESTED** [review]
Thanks, have some fixes around variable names, and questions on when we are using ErrorS vs InfoS.

File: staging/src/k8s.io/legacy-cloud-providers/aws/aws.go [hunk 8]
```diff
@@ -1466,7 +1466,7 @@ func (c *Cloud) NodeAddresses(ctx context.Context, name types.NodeName) ([]v1.No
 			}
 			num, err := strconv.Atoi(strings.TrimSpace(numStr))
 			if err != nil {
-				klog.Warningf("Bad device-number %q for interface %s\n", numStr, macID)
+				klog.InfoS("Bad device-number",numStr,"interface", macID)
```
Comment (yangjunmyfm192085) [inline staging/src/k8s.io/legacy-cloud-providers/aws/aws.go:1469]: please ensure gofmt.
run hack/update-gofmt.sh and check hack/verify-gofmt.sh.

**Pull request closed without merging**